corgi/tasks/management/commands/loadunprocessedrelations.py

```diff
@@ -7,19 +7,111 @@
 class Command(BaseCommand):
     help = "Fetch unprocessed builds from the relations table"
 
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "-p",
+            "--product_ref",
+            dest="product_ref",
+            help="Fetch unprocessed relations for product reference",
+        )
+        parser.add_argument(
+            "-t",
+            "--relation_type",
+            dest="relation_type",
+            help="Fetch unprocessed relations for type",
+        )
+        parser.add_argument(
+            "-f",
+            "--force",
+            action="store_true",
+            help="Force loading even if build exists",
+        )
+
     def handle(self, *args, **options):
-        self.stdout.write(
-            self.style.SUCCESS(
-                "Loading unprocessed relations",
+        if options["force"]:
+            response = input(
+                "Using the force option could potentially put many jobs into the slow"
+                " queue, continue? y/n"
             )
-        )
-        processed_builds = 0
-        for relation_type in ProductComponentRelation.objects.values_list(
-            "type", flat=True
-        ).distinct():
-            processed_builds += fetch_unprocessed_relations(relation_type=relation_type)
+            if response.lower() != "y":
+                exit(0)
+
+        if options["product_ref"] and options["relation_type"]:
+            self.check_product_ref(options)
+            self.check_relation_type(options)
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f"Loading relations with product ref: {options['product_ref']},"
+                    f" and type {options['relation_type']}",
+                )
+            )
+            processed_builds = fetch_unprocessed_relations(
+                product_ref=options["product_ref"],
+                relations_type=options["relation_type"],
```

Comment:
Looks like SonarQube is complaining about a typo here:
```suggestion
                relation_type=options["relation_type"],
```

Reply:
I removed this commit from the PR